admin/class-accessible-table.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ * WPSEO plugin file.
+ *
+ * @package WPSEO\Admin\Links
+ */
+
+/**
+ * Represents the state of a table being accessible.
+ */
+abstract class WPSEO_Accessible_Table {
+
+	const ACCESSIBLE   = '0';
+	const INACCESSBILE = '1';
+
+	/**
+	 * The storage class used to check the accessible table.
+	 *
+	 * @var string
+	 */
+	public static $storage_class = 'WPSEO_Link_Storage';
+
+	/**
+	 * The transient used to store the accessible state.
+	 *
+	 * @var string
+	 */
+	public static $transient_name = 'wpseo_link_table_inaccessible';
+
+	/**
+	 * Checks if the given table name exists.
+	 *
+	 * @return bool True when table is accessible.
+	 */
+	public static function is_accessible() {
+		$value = get_transient( self::transient_name() );
+
+		// If the value is not set, check the table.
+		if ( false === $value ) {
+			return self::check_table();
+		}
+
+		return $value === self::ACCESSIBLE;
+	}
+
+	/**
+	 * Sets the transient value to 1, to indicate the table is not accessible.
+	 *
+	 * @return void
+	 */
+	public static function set_inaccessible() {
+		set_transient( self::transient_name(), self::INACCESSBILE, HOUR_IN_SECONDS );
+	}
+
+	/**
+	 * Removes the transient.
+	 *
+	 * @return void
+	 */
+	public static function cleanup() {
+		delete_transient( self::transient_name() );
+	}
+
+	/**
+	 * Sets the transient value to 0, to indicate the table is accessible.
+	 *
+	 * @return void
+	 */
+	protected static function set_accessible() {
+		/*
+		 * Prefer to set a 0 timeout, but if the timeout was set before WordPress will not delete the transient
+		 * correctly when overridden with a zero value.
+		 *
+		 * Setting a YEAR_IN_SECONDS instead.
+		 */
+		set_transient( self::transient_name(), self::ACCESSIBLE, YEAR_IN_SECONDS );
+	}
+
+	/**
+	 * Checks if the table exists if not, set the transient to indicate the inaccessible table.
+	 *
+	 * @return bool True if table is accessible.
+	 */
+	protected static function check_table() {
+		global $wpdb;
+
+		$storage = new self::$storage_class();
```

Comment:
Wouldn't it be nicer to force a method in this abstract that returns an instance of a storage class:
```
abstract public function get_storage_class();
```

If you choose for this, we might consider having an interface that forces the `get_table_name` function.